frame/contracts/proc-macro/src/lib.rs

```diff
@@ -201,16 +213,31 @@ impl HostFn {
 			_ => Err(err(span, msg)),
 		}?;
 
+		// process arguments: The first and second arg are treated differently (ctx, memory)
+		// hhey must exist and be `ctx: _` and `memory: _`.
+		let msg = "Every function must start with two inferred parameters: ctx: _ and memory: _";
```

Comment:
This would be a cyclic life time between ctx and memory. We also can't just use the `Memory` in the user state type (`Runtime` in our case) because it needs a `&mut CallContext` which we already borrowed to access `Runtime`.